app/src/main/res/layout/fragment_onboarding.xml

```diff
@@ -4,26 +4,44 @@
     android:layout_width="match_parent"
     android:layout_height="match_parent">
 
-    <TextView
-        android:id="@+id/onboardingTitleTV"
-        style="@style/TextAppearance.MaterialComponents.Headline4"
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:text="Onboarding Fragment"
-        android:textAlignment="center"
-        app:layout_constraintBottom_toBottomOf="parent"
+    <androidx.viewpager2.widget.ViewPager2
+        android:id="@+id/onboardingVP2"
+        android:layout_width="match_parent"
+        android:layout_height="0dp"
+        android:orientation="horizontal"
         app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHeight_default="percent"
+        app:layout_constraintHeight_percent="0.8"
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toTopOf="parent" />
 
-    <Button
-        android:id="@+id/onboardingBtn"
+    <com.google.android.material.button.MaterialButton
+        android:id="@+id/nextItemBtn"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:layout_marginTop="16dp"
-        android:text="Get Started"
-        android:textSize="16sp"
-        app:layout_constraintEnd_toEndOf="parent"
+        android:layout_marginEnd="24dp"
+        android:layout_marginBottom="74dp"
+        android:backgroundTint="@color/accent_pink"
+        android:drawableRight="@drawable/ic_arrow_right"
+        android:text="Next"
+        android:textColor="@color/white"
+        app:cornerRadius="8dp"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent" />
+
+    <com.tbuonomo.viewpagerdotsindicator.WormDotsIndicator
+        android:id="@+id/dots_indicator"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="24dp"
+        android:layout_marginBottom="98dp"
+        app:dotsColor="@color/accent_pink"
```

Comment:
Create separate color resources to be called outside `colors.xml`
This way if you need to implement dark mode, you can have a colors file for dark mode with the exact same names for colors, and only the specific colors like white purple might be different

For example have a color res `text_color` that references white and in the dark mode colors file you'll have the same `text_color` res name but it might reference red. This way you only have to change the dark mode colors instead of everywhere in your code.

I suggest applying this change wherever necessary 😁

On a real life project this saves a lot of development time if the client decides they want dark mode after the initial release

Reply:
I'm not sure I understand. This means that I should create a new res file to hold my custom colors?